### PR TITLE
Add Vulkan Cube to GAPID apk.

### PR DESCRIPTION
**Pull request by lpy**
In order to use Vulkan Cube to run device validation, integrate it into gapid
apk.

BUG: b/138717619
Test: bazel run gapit validate_gpu_profiling

### CODE REVIEW

**Comment by lpy** [comment]
ptal

File: gapis/trace/android/validate/cc/cube.h [hunk 1]
```diff
@@ -0,0 +1,154 @@
+#pragma once
```
Comment (hevrard) [inline gapis/trace/android/validate/cc/cube.h]: Missing copyright notice. This is probably from Vulkan SDK, and if so should be under `third_party`

File: gapis/trace/android/validate/cc/gettime.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/**************************************************************************
+ *
+ * Copyright 2014, 2017 The Android Open Source Project
```
Comment (hevrard) [inline gapis/trace/android/validate/cc/gettime.h]: Again, anything not (c) Google is meant to be under third_party

File: gapis/trace/android/validate/cc/native-lib.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#include <android/asset_manager.h>
```
Comment (hevrard) [inline gapis/trace/android/validate/cc/native-lib.cpp]: Missing copyright

File: gapis/trace/android/validate/cc/vulkan_helper.cpp [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#include "vulkan_helper.h"
```
Comment (hevrard) [inline gapis/trace/android/validate/cc/vulkan_helper.cpp]: Copyright

File: gapis/trace/android/validate/cc/vulkan_helper.h [hunk 5]
```diff
@@ -0,0 +1,119 @@
+#pragma once
```
Comment (hevrard) [inline gapis/trace/android/validate/cc/vulkan_helper.h]: Copyright

**Comment by lpy** [comment]
As suggested by Hugues, I am moving this to a third_party folder.

**Comment by lpy** [comment]
Moved the everything to `third_party`

File: tools/build/workspace.bzl [hunk 6]
```diff
@@ -278,7 +278,7 @@ def gapid_dependencies(android = True, mingw = True, locals = {}):
             native.android_ndk_repository,
             name = "androidndk",
             locals = locals,
-            api_level = 23, # This is the minimum API
+            api_level = 26, # This is the minimum API
```
Comment (AWoloszyn) [inline tools/build/workspace.bzl]: @pmuetschard  Can we actually make this change? Does this break other things?

File: gapidapk/android/apk/AndroidManifest.xml.in [hunk 7]
```diff
@@ -24,6 +24,9 @@
         android:minSdkVersion="21"
         android:targetSdkVersion="26" />
 
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
+    <uses-permission android:name="android.permission.INTERNET" />
```
Comment (pmuetschard) [inline gapidapk/android/apk/AndroidManifest.xml.in]: Neither of these should be required. If the cube needs them, it should be changed to not need them.

File: tools/build/third_party/validate/assets/cube.vert [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+/*
+ * Vertex shader used by Cube demo.
+ */
+#version 400
+#extension GL_ARB_separate_shader_objects : enable
+#extension GL_ARB_shading_language_420pack : enable
+layout(std140, binding = 0) uniform buf {
+        mat4 MVP;
+        vec4 position[12*3];
+        vec4 attr[12*3];
```
Comment (pmuetschard) [inline tools/build/third_party/validate/assets/cube.vert]: disappointing that we are using this very corner-casey way of specifying geometry. How do we know that this won't confuse the driver too much and put it into a non-standard rendering mode?

File: tools/build/third_party/validate/cc/BUILD.bazel [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+load("//tools/build:rules.bzl", "android_dynamic_library", "cc_copts")
+
+cc_library(
+    name = "cc",
+    srcs = glob(["*.cpp"]),
+    hdrs = glob(["*.h"]),
+    copts = cc_copts() + select({
+        # Android
+        "//conditions:default": ["-DVK_USE_PLATFORM_ANDROID_KHR"],
+    }) + [
```
Comment (pmuetschard) [inline tools/build/third_party/validate/cc/BUILD.bazel]: if the only case in the select is the default case, what's the point of the select?

File: tools/build/third_party/validate/cc/BUILD.bazel [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+        "-std=c++17",
+        "-fno-rtti",
+        "-fno-exceptions",
+        "-DNDEBUG",  # always ndebug for perfetto
```
Comment (pmuetschard) [inline tools/build/third_party/validate/cc/BUILD.bazel]: perfetto?

**Pull request closed without merging**